tulip/synth.py

```diff
@@ -980,28 +980,42 @@ def synthesize_many(specs, ts=None, ignore_init=None,
     assert isinstance(ts, dict)
     for name, t in ts.items():
         assert isinstance(t, transys.FiniteTransitionSystem)
-        ignore = name in ignore_init
-        bool_act = name in bool_actions
+        ignore = ignore_init.get(name) 
+        bool_act = bool_actions.get(name)
         statevar = name
         if t.owner == 'sys':
-            specs |= sys_to_spec(t, ignore, statevar,
+            sys_spec = sys_to_spec(t, ignore, statevar,
                                  bool_actions=bool_act)
+            sys_spec.qinit = specs.qinit 
+            sys_spec.moore = specs.moore 
+            sys_spec.plus_one = specs.plus_one 
+            specs |= sys_spec
         elif t.owner == 'env':
-            specs |= env_to_spec(t, ignore, statevar,
+            env_spec = env_to_spec(t, ignore, statevar,
                                  bool_actions=bool_act)
+            env_spec.qinit = specs.qinit 
+            env_spec.moore = specs.moore 
+            env_spec.plus_one = specs.plus_one 
+            specs |= env_spec
     if solver == 'gr1c':
-        ctrl = gr1c.synthesize(specs)
+        strategy = gr1c.synthesize(specs)
     elif solver == 'slugs':
         if slugs is None:
             raise ValueError(
                 'Import of slugs interface failed. '
                 'Please verify installation of "slugs".')
-        ctrl = slugs.synthesize(specs)
+        strategy = slugs.synthesize(specs)
     else:
         raise Exception((
             'Unknown solver: "{solver}". '
             'Available solvers: "gr1c", and "slugs"').format(
                 solver=solver))
+    # While the return values of the solver interfaces vary, we expect
+    # here that strategy is either None to indicate unrealizable or a
+    # networkx.DiGraph ready to be passed to strategy2mealy().
+    if strategy is None:
+        return None
+    ctrl = strategy2mealy(strategy, specs)
```

Comment:
The error here was a missing conversion from an enumerated state graph with the synthesized transitions to a Mealy machine. This error appears to have been introduced in 74807640a1a1fe6da2e579189dc5ce4a0b0ea118. The conversion was previously within the module `tulip.interfaces.gr1c`. When the conversion was moved to `tulip.synth` (this module), the function `synthesize` was updated, but the function `synthesize_many` was forgotten, which introduced the error.

Moreover, `omega` and `gr1py` are missing from the synthesizer options available in this function. I think that the only reason they are missing is that we added them to `synthesize`, but forgot to add them to `synthesize_many`. If so, then we should add them in `synthesize_many` too.

Also, the added lines are identical with the relevant area in `synthesize`:

https://github.com/tulip-control/tulip-control/blob/1ad5b12ce1f96253d898abb8485d4e0ca8d20d90/tulip/synth.py#L1140-L1145